sklearn/mixture/gmm.py

```diff
@@ -448,14 +455,20 @@ def fit(self, X):
             log_likelihood = []
             # reset self.converged_ to False
             self.converged_ = False
+
+            # this line should be removed when 'thresh' is deprecated
+            tol = self.tol if self.thresh is None \
+                else self.thresh / float(X.shape[0])
+
             for i in range(self.n_iter):
                 # Expectation step
                 curr_log_likelihood, responsibilities = self.score_samples(X)
-                log_likelihood.append(curr_log_likelihood.sum())
+                log_likelihood.append(curr_log_likelihood.mean())
```

Comment:
This change should be documented in `whats_new.rst` and motivated by the fact that it makes the stopping condition less dependent on the number of samples.

@bthirion @agramfort @GaelVaroquaux any comment on that particular point?